build-tools/packages/build-cli/src/handlers/checkFunctions.ts

```diff
@@ -447,6 +439,11 @@ export const checkPolicy: StateHandlerFunction = async (
             );
             BaseStateHandler.signalFailure(machine, state);
         }
+        // eslint-disable-next-line no-negated-condition
+    } else if (!branchesRunDefaultPolicy.includes(context.originalBranchName)) {
+        log.verbose(
+            `Skipping policy check because it does not run on the ${context.originalBranchName} branch by default. Pass --policyCheck to force it to run.`,
```

Comment:
Well, it could be skipped because you passed `--no-policycheck` or `skipFlags`, right?  Probably fine since you would probably know what you're doing if you're passing those flags?

You could consider updating `FluidReleaseStateHandlerData.shouldCheckPolicy` to be a more expressive type modeling all the interesting combos of inputs.  Or not 😅